### PR TITLE
[plugin.video.vrt.nu@matrix] 2.3.3+matrix.1

### PR DESCRIPTION
**Pull request by mediaminister**
### Add-on details:

- **General**
  - Add-on name: VRT NU
  - Add-on ID: plugin.video.vrt.nu
  - Version number: 2.3.3+matrix.1
  - Kodi/repository version: matrix

- **Code location**
  - URL: https://github.com/add-ons/plugin.video.vrt.nu
  
VRT NU is the video-on-demand platform of the Flemish public broadcaster (VRT).

  - Track the programs you like
  - List all videos alphabetically by program, category, channel or feature
  - Watch live streams from Eén, Canvas, Ketnet, Ketnet Junior and Sporza
  - Discover recently added or soon offline content
  - Browse the online TV guides or search VRT NU

[I]The VRT NU add-on is not endorsed by VRT, and is provided 'as is' without any warranty of any kind.[/I]

### Description of changes:

v2.3.3 (2020-04-30)
- Fix Radio 1 livestream
- Fix watching VRT NU abroad
- Add support for Kodi 19 Matrix "pre-release" builds
- Add colour preferences to accomodate light skins
- Add a favourites music subsection
- Allow editing a search query from search history
- Expand TV guide range
- Add livestreams for Radio 2 and Klara

v2.3.2 (2020-01-18)
- Fix issues related to categories scraping and caching
- Various improvements to TV guide

v2.3.1 (2020-01-14)
- Fix an issue related to type-handling for settings
- Fix an issue related to marking played videos as watched when using Up Next
- Small fix for livestreams
- Improvements to metadata handling specific to Kodi rendering
- Support multiple YouTube links per channel

v2.3.0 (2020-01-01)
- Cosmetic changes to metadata and plot
- Store resume points and watchlater at VRT NU
- Implement "Watch later"
- Implement "Continue watching"
- Add integration for Up Next add-on
- Fixes to "Go to program", "TV guide" and "Live TV" entries
- Fix Python 3.5 support

v2.2.3 (2019-10-01)
- Integrate new inputstreamhelper
- Support user paths with unicode characters
- Manage My favorites
- Improve breadcrumbs support
- Various translation fixes

v2.2.2 (2019-09-01)
- "Go to program" will now got to "All seasons" listing
- Rename "My programs" to "My favorites"
  

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

**Comment by basrieter** [comment]
Discussing https://github.com/xbmc/repo-plugins/pull/2872/

**Comment by dagwieers** [comment]
Approved by maintainer.

File: plugin.video.vrt.nu/resources/lib/kodiutils.py [hunk 1]
```diff
@@ -0,0 +1,1102 @@
...
+        if title_item.context_menu:
+            list_item.addContextMenuItems(title_item.context_menu)
+
+        url = None
+        if title_item.path:
+            url = title_item.path
+
+        listing.append((url, list_item, is_folder))
+
+    # Jump to specific item
```
Comment (enen92) [inline plugin.video.vrt.nu/resources/lib/kodiutils.py:236]: consider removing later

File: plugin.video.vrt.nu/resources/lib/kodiutils.py [hunk 2]
```diff
@@ -0,0 +1,1102 @@
...
+    elif proxy.get('server'):
+        proxy_address = '{scheme}://{server}'.format(**proxy)
+    else:
+        return None
+
+    return dict(http=proxy_address, https=proxy_address)
+
+
+def get_cond_visibility(condition):
+    """Test a condition in XBMC"""
```
Comment (enen92) [inline plugin.video.vrt.nu/resources/lib/kodiutils.py:633]: kodi

File: plugin.video.vrt.nu/resources/lib/kodiutils.py [hunk 3]
```diff
@@ -0,0 +1,1102 @@
...
+
+
+def wait_for_resumepoints():
+    """Wait for resumepoints to be updated"""
+    update = get_property('vrtnu_resumepoints')
+    if update == 'busy':
+        import time
+        timeout = time.time() + 5  # 5 seconds timeout
+        log(3, 'Resumepoint update is busy, wait')
+        while update != 'ready':
```
Comment (enen92) [inline plugin.video.vrt.nu/resources/lib/kodiutils.py:852]: Consider checking for abortequested here

**Pull request merged**